docs/kms_secretlock.md

```diff
@@ -0,0 +1,143 @@
+# Aries New Secure KMS, SecretLock and Crypto capabilities
+
+The aries-framework-go project has a mandate to protect keys from being compromised during usage or storage of keys. 
+This goal can achieved by hiding raw key bytes from being exchanged between the user and the framework. Also, keys
+should be secured and locked by a master key that is knwon to the user and protected by a passphrase. In order 
+to achieve this, the following criteria must be met:
+ - The KMS service should support providing a key reference (or keyset handle) whenever a key is required. It must not
+   return or accept raw contents of keys within the framework and with the user.
+ - A SecretLock service is needed to protect keys by encrypting them when being stored.
+ - The SecretLock service must work with a master key that can be provided by the user.
+ - The master key should also be encrypted to protect it from being compromised.
+ - It should be unlocked by a passphrase known to the user.
```

Comment:
Again - passphrase is only one of the mechanisms.

Reply:
updated to reflect this as an example